internal/driver/node.go

```diff
@@ -363,22 +363,11 @@ func (s *nodeServerNoLocked) NodeUnpublishVolume(ctx context.Context, req *csi.N
 func (s *nodeServerNoLocked) nodeUnpublishFilesystemVolume(req *csi.NodeUnpublishVolumeRequest, device string) error {
 	targetPath := req.GetTargetPath()
 
-	mounted, err := filesystem.IsMounted(device, targetPath)
-	if err != nil {
-		return status.Errorf(codes.Internal, "mount check failed: target=%s, error=%v", targetPath, err)
-	}
-	if mounted {
-		if err := s.mounter.Unmount(targetPath); err != nil {
-			return status.Errorf(codes.Internal, "unmount failed for %s: error=%v", targetPath, err)
-		}
-	}
-
-	if err := os.RemoveAll(targetPath); err != nil {
-		return status.Errorf(codes.Internal, "remove dir failed for %s: error=%v", targetPath, err)
+	if err := mountutil.CleanupMountPoint(targetPath, s.mounter, true); err != nil {
```

Comment:
Let me double check but I believe this was by mistake if this happened.

Reply:
The current behavior is 

```go
pathExists, pathErr := PathExists(mountPath)
	if !pathExists && pathErr == nil {
		klog.Warningf("Warning: mount cleanup skipped because path does not exist: %v", mountPath)
		return nil
	}
```
This means that if we have a non existing path we should be fine. Also If there is no link present we will run into another nil case that is covered with IsCorruptedMount. This should be covering the cases we had before.

Reply:
Thank you for your explanation. I understood.